libs/ui-lib/lib/ocm/components/clusterConfiguration/ExternalPlatformDropdown.tsx

```diff
@@ -15,62 +16,112 @@ import {
   VSPHERE_CONFIG_LINK,
   getFieldId,
 } from '../../../common';
+import { ExternalPlatformIntegrationStateMap, FeatureIdPlatform } from './types';
 
 const INPUT_NAME = 'platform';
 const fieldId = getFieldId(INPUT_NAME, 'input');
 export type ExternalPlatformType = Extract<PlatformType, 'none' | 'nutanix' | 'oci' | 'vsphere'>;
 
 type ExternalPlatformDropdownProps = {
   showOciOption: boolean;
-  disabledOciTooltipContent: React.ReactNode;
-  isOciDisabled: boolean;
   onChange: (selectedPlatform: ExternalPlatformType) => void;
-  dropdownIsDisabled: boolean;
+  cpuArchitecture?: string;
+  clusterExists?: boolean;
+  clusterPlatform?: PlatformType;
+  externalPlatformIntegrationStateMap: ExternalPlatformIntegrationStateMap;
 };
 
 export type ExternalPlatformInfo = {
   label: string;
   href: string;
   tooltip: string;
+  featureId: string;
 };
 
 export const externalPlatformTypes: Record<ExternalPlatformType, ExternalPlatformInfo> = {
   none: {
     label: 'No platform integration',
     href: '',
     tooltip: '',
+    featureId: '',
   },
   nutanix: {
     label: 'Nutanix',
     href: NUTANIX_CONFIG_LINK,
     tooltip: '',
+    featureId: 'NUTANIX_INTEGRATION',
   },
   oci: {
     label: 'Oracle  (Requires a custom manifest)',
     href: '',
     tooltip:
       "To integrate with an external partner (for example, Oracle Cloud), you'll need to provide a custom manifest.",
+    featureId: 'EXTERNAL_PLATFORM_OCI',
   },
   vsphere: {
     label: 'vSphere',
     href: VSPHERE_CONFIG_LINK,
     tooltip: '',
+    featureId: 'VSPHERE_INTEGRATION',
```

Comment:
As a general approach, if each one of the entries in this object implements a similar interface, it would be better to have a "maker" function that instantiates objects with this same interface.
For example:
```typescript
const makeExternalPlatformInfo = (options: ExternalPlatformInfo = {}): ExternalPlatformInfo => ({
    label: options.label ?? '',
    href: options.href ?? '',
    tooltip: options.tooltip ?? '' ,
    featureId: options.featureId ?? '',
  })
```